generator/test/good_test_data/all_types.expected

```diff
@@ -0,0 +1,221 @@
+// **************************************************************************
+// RealmObjectGenerator
+// **************************************************************************
+
+class Fooo extends _Foo with RealmEntity, RealmObject {
```

Comment:
Why does this have 3 `o`-s? Shouldn't the generated class be called `Foo`?

Reply:
We support a map to directive on the model class. Which in  this case is like 
@RealmModel()
**@MapTo('Fooo')**
class _Foo {
  int x = 0;
}

Reply:
I agree this exact sample is not ideal. should have been named differently.

Reply:
I pushed a change on that name to avoid the confusion https://github.com/realm/realm-dart/commit/8d04e3f69d9392d9e090b5db9944f7a3d9ae172c